internal/pr/promote/github_pull_request_provider.go

```diff
@@ -112,6 +110,44 @@ func (g *GitHubPullRequestProvider) SetPromotionEnvironment(branch string, env s
 	return nil
 }
 
+func get(branch string) (*pullRequest, error) {
+	// List pull requests for branch
+	cmd := exec.Command("gh", "pr", "list", "--head", branch, "--json", "headRefName,labels")
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return nil, fmt.Errorf("listing pull requests for branch %s: %s", branch, output)
+	}
+
+	// Unmarshal JSON
+	var prs []pullRequest
+	if err := json.Unmarshal(output, &prs); err != nil {
+		return nil, fmt.Errorf("unmarshaling pull request list: %w", err)
+	}
+
+	// We can safely assume that there is either none or only one PR for a given branch
+	if len(prs) == 0 {
+		return nil, nil
+	}
+	return &prs[0], nil
+}
+
+func getAll() ([]pullRequest, error) {
+	// List pull requests for branch
+	cmd := exec.Command("gh", "pr", "list", "--json", "headRefName,labels")
```

Comment:
On older/more active repos, we are going to hit paging issues. We can filter out by labels:

```
$ gh pr list --label 'promote:<env>' --state open
```

We might want to get merged or closed PRs but remove the labels without prompting the user.

Reply:
I addressed that, thanks! :)